components/sessions/content/brave_content_serialized_navigation_driver_unittest.cc

```diff
@@ -0,0 +1,45 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "components/sessions/content/content_serialized_navigation_driver.h"
+
+#include "components/sessions/core/serialized_navigation_entry.h"
+#include "components/sessions/core/serialized_navigation_entry_test_helper.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+namespace sessions {
```

Comment:
I'm not sure a unit test is sufficient for this. I'd like to see that chrome:// pages restore correctly, in particular ones that contain text areas (adblock settings?)

Reply:
Briefly building and trying this out, it doesn't seem to affect restoring (ex: closing tab and using Ctrl +Shift + T). Changes are saved just fine too

The serializing happen is when it goes to disk I believe. I don't think we want these values persisting to the autofill. Maybe a better example to try would be if there is a page we DO want autofill to save on

Reply:
it is the session restore feature that saves PageState to disk periodically and not related to the autofill. PageState internaly skips `input`s with `autocomplete=off` which can be confusing.

Reply:
Added a browser test to make sure we do not serialize data from `chrome://` pages